# Corrigé Exercice 1 (Module 02) : Chargement et exploration

## Partie 1

In [1]:
import pandas as pd

df = pd.read_csv("../../assets/datasets/entreprises_panel.csv")


## Partie 2

In [2]:
df.head(10)


,entreprise_id,annee,secteur,region,annee_creation,teletravail,effectifs,chiffre_affaires_keur,rentabilite,satisfaction_rh
0,E001,2019,Industrie,Occitanie,2018,1,529,115830.2,0.1706,7.7
1,E001,2020,Industrie,Occitanie,2018,1,590,125110.5,0.1497,6.7
2,E001,2021,Industrie,Occitanie,2018,1,610,123760.0,0.1594,5.5
3,E001,2022,Industrie,Occitanie,2018,1,652,127441.6,0.1703,8.2
4,E001,2023,Industrie,Occitanie,2018,1,731,140627.0,0.0057,7.0
5,E002,2019,Technologies,Nouvelle-Aquitaine,2009,1,381,75355.0,0.1907,8.0
6,E002,2020,Technologies,Nouvelle-Aquitaine,2009,1,379,72634.8,0.1020,5.4
7,E002,2021,Technologies,Nouvelle-Aquitaine,2009,1,378,77716.4,0.1693,8.0
8,E002,2022,Technologies,Nouvelle-Aquitaine,2009,1,346,83106.0,0.1043,7.7
9,E002,2023,Technologies,Nouvelle-Aquitaine,2009,1,384,73128.4,0.0645,7.9


## Partie 3

In [3]:
# 1. Nombre de lignes et de colonnes
df.shape


(404, 10)

In [4]:
# 2. Moyenne du chiffre d'affaires
df["chiffre_affaires_keur"].mean()


np.float64(41128.63094059406)

In [5]:
# 3. Valeur maximale de la rentabilité
df["rentabilite"].max()


np.float64(0.2707)

## Partie 4

In [6]:
df.info()


<class 'pandas.DataFrame'>
RangeIndex: 404 entries, 0 to 403
Data columns (total 10 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   entreprise_id          404 non-null    str    
 1   annee                  404 non-null    int64  
 2   secteur                404 non-null    str    
 3   region                 404 non-null    str    
 4   annee_creation         404 non-null    int64  
 5   teletravail            404 non-null    int64  
 6   effectifs              404 non-null    int64  
 7   chiffre_affaires_keur  404 non-null    float64
 8   rentabilite            404 non-null    float64
 9   satisfaction_rh        396 non-null    float64
dtypes: float64(3), int64(4), str(3)
memory usage: 31.7 KB


**Réponse** : la colonne `satisfaction_rh` contient des valeurs manquantes (on le voit car son `Non-Null Count` est inférieur au nombre total de lignes, 404). On peut obtenir le chiffre exact avec `df["satisfaction_rh"].isna().sum()`.


In [7]:
df["satisfaction_rh"].isna().sum()


np.int64(8)

## Partie 5

In [8]:
# 1. Colonne secteur seule (une Series)
secteur = df["secteur"]
print(type(secteur))
secteur.head()


<class 'pandas.Series'>


0    Industrie
1    Industrie
2    Industrie
3    Industrie
4    Industrie
Name: secteur, dtype: str

In [9]:
# 2. Trois colonnes ensemble (un DataFrame)
sous_tableau = df[["entreprise_id", "annee", "secteur"]]
print(type(sous_tableau))
sous_tableau.head()


<class 'pandas.DataFrame'>


,entreprise_id,annee,secteur
0,E001,2019,Industrie
1,E001,2020,Industrie
2,E001,2021,Industrie
3,E001,2022,Industrie
4,E001,2023,Industrie
